logic/src/androidMain/kotlin/com/wire/kalium/logic/sync/WorkSchedulerImpl.kt

```diff
@@ -88,101 +98,171 @@ class WrapperWorkerFactory(private val coreLogic: CoreLogic) : WorkerFactory() {
-            .putSerializable(USER_ID_KEY, userId)
+            .apply { if (userId != null) putSerializable(USER_ID_KEY, userId) }
             .build()
     }
 }
 
-actual class WorkSchedulerImpl(private val context: Context, private val userId: UserId) : WorkScheduler {
+actual sealed class WorkSchedulerImpl(private val appContext: Context) : WorkScheduler {
+    internal val workerClass = WrapperWorker::class.java
 
-    private val workerClass = WrapperWorker::class.java
-    override fun enqueueImmediateWork(work: KClass<out UserSessionWorker>, name: String) {
-        val inputData = WrapperWorkerFactory
-            .workData(work, userId)
+    /**
+     * Schedules some work to be done in the background, in a "run and forget" way – free from client app observation.
+     * On mobile clients for example, aims to start a job that will not be suspended by the user minimizing the app.
+     */
+    override fun enqueueImmediateWork(work: KClass<out DefaultWorker>, name: String) {
+        val inputData = WrapperWorkerFactory.workData(work, if (this is UserSession) this.userId else null)
 
         val request = OneTimeWorkRequest.Builder(workerClass)
             .setExpedited(OutOfQuotaPolicy.RUN_AS_NON_EXPEDITED_WORK_REQUEST)
             .setInputData(inputData).build()
 
-        WorkManager.getInstance(context).beginUniqueWork(
+        WorkManager.getInstance(appContext).beginUniqueWork(
             name,
             ExistingWorkPolicy.REPLACE,
             request
         ).enqueue()
     }
 
-    override suspend fun scheduleSendingOfPendingMessages() {
-        val inputData = WrapperWorkerFactory.workData(PendingMessagesSenderWorker::class, userId)
+    actual class Global(
+        private val appContext: Context
+    ) : WorkSchedulerImpl(appContext), GlobalWorkScheduler {
 
-        val connectedConstraint = Constraints.Builder()
-            .setRequiredNetworkType(NetworkType.CONNECTED)
-            .build()
+        override fun schedulePeriodicApiVersionUpdate() {
+            val inputData = WrapperWorkerFactory.workData(UpdateApiVersionsWorker::class)
 
-        val request = OneTimeWorkRequest.Builder(workerClass)
-            .setConstraints(connectedConstraint)
-            .setInputData(inputData)
-            .build()
+            val connectedConstraint = Constraints.Builder()
+                .setRequiredNetworkType(NetworkType.CONNECTED)
+                .build()
 
-        WorkManager.getInstance(context).enqueueUniqueWork(
-            WORK_NAME_PREFIX_PER_USER + userId.value,
-            ExistingWorkPolicy.APPEND,
-            request
-        )
-    }
+            val scheduledHourOfDayToExecute = TIME_OF_EXECUTION // schedule at 4AM
+            val repeatIntervalInHours = REPEAT_INTERVAL // execute every 24 hours
+            val localTimeZone = TimeZone.currentSystemDefault()
+            val timeNow: Instant = Clock.System.now() // current time
```

Comment:
Good question, I can move those values to the contractor with default values, so it can be changed for tests and for the case if the worker is part of another class then mocking the result will do the trick.

Reply:
We can mock it, I did it in one of our tests as I remember.